src/hotspot/os_cpu/windows_x86/orderAccess_windows_x86.hpp

```diff
@@ -59,8 +60,12 @@ inline void OrderAccess::fence() {
 }
 
 inline void OrderAccess::cross_modify_fence_impl() {
-  int regs[4];
-  __cpuid(regs, 0);
+  if (VM_Version::supports_serialize()) {
+    _serialize();
```

Comment:
I cannot see this defined in the list of intrinsics available for Visual Studio:

https://docs.microsoft.com/en-us/cpp/intrinsics/x64-amd64-intrinsics-list?view=msvc-160

Reply:
I was compiling using --with-toolchain-version=2019
_serialize() is defined here https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=seri&expand=6062

Reply:
Also ran specjvm2008 successfully

Reply:
You are linking to Intel documentation of the Intel C++ compiler. This `_serialize()` intrinsic needs to be available in Microsoft Visual Studio.

Reply:
I ran it through our build system and it builds fine but I'd be happier if I could find some documentation that shows the availability of `_serialize()` in Visual Studio.

Reply:
Update: it builds fine with VS 2019, but not VS 2017. I don't know if we still expect to be able to build with VS 2017. If we do then the new code will need a compiler version guard on it.

Reply:
Update: VS 2017 is the minimum supported compiler version so the new code will need to be enabled only for VS 2019.